packages/react/src/jsx/ReactJSXElement.js

```diff
@@ -340,12 +344,14 @@ export function jsxProd(type, config, maybeKey) {
     }
   }
 
-  // Resolve default props
-  if (type && type.defaultProps) {
-    const defaultProps = type.defaultProps;
-    for (propName in defaultProps) {
-      if (props[propName] === undefined) {
-        props[propName] = defaultProps[propName];
+  if (!disableDefaultPropsExceptForClasses) {
+    // Resolve default props
```

Comment:
Shouldn't this be removed even when the flag is off? Because the resolution moves regardless?

Reply:
The way I implemented the flag is that when it's off, it maintains the old behavior: props resolved during JSX. I did it that way to aid the roll out at Meta and reduce the risk the whole change gets reverted. Since there could be code that reads `element.props` and expects the `defaultProps` to already be there.

Reply:
So really the flag handles both moving the resolution and removing it for non-classes, I just didn't think it was worth creating two separate flags.